pkg/backend/filestate/state.go

```diff
@@ -242,25 +239,27 @@ func (b *localBackend) backupStack(name tokens.QName) error {
 
 	// Read the current checkpoint file. (Assuming it aleady exists.)
 	stackPath := b.stackPath(name)
-	byts, err := ioutil.ReadFile(stackPath)
+	byts, err := b.bucket.ReadAll(context.Background(), stackPath)
 	if err != nil {
 		return err
 	}
 
 	// Get the backup directory.
 	backupDir := b.backupDirectory(name)
 
-	// Ensure the backup directory exists.
-	if err = os.MkdirAll(backupDir, 0700); err != nil {
-		return err
+	// Ensure the backup directory exists (only required for local filesystem).
+	if b.urlScheme() == "file://" {
```

Comment:
Is this needed? I believe `fileblob` will do it for you as part of the write:
https://github.com/google/go-cloud/blob/master/blob/fileblob/fileblob.go#L435

Reply:
Good point.  Have also removed from https://github.com/pulumi/pulumi/pull/2455/files#diff-d335e4296a9a5904243d0f16ea2801a1R109